### PR TITLE
Schema split

### PR DESCRIPTION
**Pull request by JohnAlbin**
For my next trick, I'd like to split the schema and mocked data into multiple files.

~~One thing I don't like about the current implementation is that the Star Wars mocked data is a well-known fixed set. But we will never have a project like that. We will most likely want random-ish looking data and the Star Wars data we have is a fixed JS array. A more useful example would use Faker for all fields.)~~ [edit: Philipp makes a good argument that most of our tests will be using specific data for edge case testing. While it's easy to add edge cases to randomized data, we can revisit this issue later. Lagoon uses randomized data + edge case modifications inside tests. Our latest AL project will be using all edge case data. We can compare and contrast in a few months.]

Also, I moved the old files out of `/mocks` because there is more than just mock data in those files. The entire graphQL schema definition is there. ~~I put the resulting files in `/src/schema` for now. Maybe just `/schema` is better? I'm unsure.~~  [Update: based on latest AL project, this PR uses `/graphql/schema`.]

### CODE REVIEW

**Comment by pmelab** [comment]
Thanks! I'm all in for grouping resolvers with their mocks and splitting up the schema. Looks great. It would also be interesting to add an example of how the implementation and testing of an actual Apollo use case would look like. E.g. a GraphQL schema that consumes a REST API and mocks it for testing the resolvers.

The topic of randomness is interesting. The probably most important part of BDD is coming up with carefully crafted but realistic examples for the different scenarios and edge cases. To the point that this data is described in its own scenarios and reused across the project to make it easier to talk about it. Instead of re-explaining `"white male human Jedi with parental issues, trained by a sentient avocado"` over and over, we can just reference `"Luke Skywalker"` and everybody is immediately on the same page.

In short, I think for communication, scenarios and overall consistency this well-known, non-random example datasets are actually very beneficial.

But I see a use case for faker data in exploratory testing. It would be great to alternatively start mocks in a "chaos monkey" mode, that fills everything with realistic random data to allow a human to manually search for unknown edge cases. Or even literally run [chaos monkey](https://www.npmjs.com/package/chaos-monkey) on it.

**Comment by JohnAlbin** [comment]
AFAIK, we aren't building the `/graphql` code into a `/graphql/buildd` directory, so there's no reason to have a `/graphql/src` directory.

Also, while the WIP version of this PR had a `/graphql/schema/swapi` directory to make it easier to delete the entire example schema, I see that it led to one project having a `/graphql/schema/PROJECTNAME` directory; that extra directory is nesting for no good reason. So I've removed that level from this PR. I would change this PR to have `/graphql/schema/swapi-TYPE` but I fear that new projects would again follow the boilerplate and have `/graphql/schema/PROJECT-TYPE`.

We are creating a single schema so theres no need to have the PROJECT name used anywhere in file or directory names. And that is regardless of how many actual endpoints we have; extra endpoints need to be schema stitched into the existing project endpoint's schema (of which there is only one.)

**Comment by pmelab** [comment]
There might be a use case for creating multiple schemes per project (we talked about separating read, write and authentication GraphQL endpoints for the sake of dedicated scaling and resilience). But in these cases, the projects just can add the split themselves. And we have projects that don't employ Apollo at runtime at all. 

*BUT* there is a use case for multiple Apollo mock schemas that don't have a production counterpart. It's an easy way to mock another gatsby source plugin (e.g. a google sheet) for running tests against it.

**Comment by pmelab** [comment]
Also, I was thinking if we should move everything into `src` and only keep entry point script files outside. What do you think about that?
